### PR TITLE
Drop nuphar

### PR DESCRIPTION
**Pull request by RandySheriffH**
Deprecate nuphar for release 1.12 and after.

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -60,7 +60,7 @@ option(onnxruntime_BUILD_OBJC "Build Objective-C library" OFF)
 option(onnxruntime_USE_PREINSTALLED_EIGEN "Use pre-installed EIGEN. Need to provide eigen_SOURCE_PATH if turn this on." OFF)
 option(onnxruntime_BUILD_BENCHMARKS "Build ONNXRuntime micro-benchmarks" OFF)
 
-option(onnxruntime_USE_NUPHAR_TVM "Build TVM for code generation for NUPHAR EP" OFF)
+# option(onnxruntime_USE_NUPHAR_TVM "Build TVM for code generation for NUPHAR EP" OFF)
```
Comment (edgchen1) [inline cmake/CMakeLists.txt:63]: general: could the commented-out code just be removed?

File: onnxruntime/test/providers/cpu/math/matmul_integer_test.cc [hunk 2]
```diff
@@ -312,62 +312,33 @@ void RunMatMulIntegerU8X8Test(const int M, const int N, const int K, bool non_ze
...
-  if (non_zero_zp) {
-    test.Run(OpTester::ExpectResult::kExpectSuccess, "", {kNupharExecutionProvider});
-  } else {
-    test.Run();
-  }
+  test.Run();
 }
 
 void RunMatMulIntegerU8X8TestBatch(const int M, const int N, const int K) {
-  RunMatMulIntegerU8X8Test<int8_t>(M, N, K, false /*non_zero_zp*/, false /*B_is_initializer*/, false /*per_column_zp*/);
```
Comment (souptc) [inline onnxruntime/test/providers/cpu/math/matmul_integer_test.cc:355]: why remove those tests?

Comment (RandySheriffH) [reply onnxruntime/test/providers/cpu/math/matmul_integer_test.cc:355]: Because when "non_zero_zp", the function will test Nuphar EP exclusively.

**Comment by RandySheriffH** [comment]
> I think we need to reach an agreement with the 1st part users first before making this change.

we have settled an agreement with them already, will forward more info offline.

**Comment by snnn** [comment]
@pranavsharma , would you please help us remove nuphar from the required pipelines? Otherwise this PR won't pass.

**Comment by snnn** [comment]
@RandySheriffH, please fix the  build errors.

**Comment by pranavsharma** [comment]
> @pranavsharma , would you please help us remove nuphar from the required pipelines? Otherwise this PR won't pass.

done

**Review by snnn — APPROVED** [review]

**Comment by pranavsharma** [comment]
@RandySheriffH @souptc  I still see nuphar files and references to nuphar in other files. Is this intentional? For e.g. https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/providers/nuphar/nuphar_provider_factory.h

**Pull request merged**